build.psm1

```diff
@@ -2215,9 +2212,12 @@ function Start-CrossGen {
         throw "crossgen is not available for this platform"
     }
 
+    $dotnetRuntimeVersion = $dotnetCLIRequiredVersion -match "(\d\.\d)\..*" | ForEach-Object { $matches[1] }
```

Comment:
This will give us the Major.Minor version of the sdk, not the runtime. The sdk version is different from the runtime version, e.g. we use sdk 2.1.4 currently with runtime version 2.0. It may happen in future that sdk moved to 2.2 while we are still using 2.1 runtime.

Reply:
Seems we need review this after .Net Core 2.1 release because it introduce new policy - always use latest version. It is still not work for Preview versions. And seems they want adjust this versions for simplicity.

Reply:
> It is still not work for Preview versions.

What is not working for preview versions?

Reply:
The problem we had in CI w/o this is that we pick up the first version that matches which was crossgen 2.2 (some preview version) which is not compatible with the 2.1 built binaries.

Reply:
I wonder how come we have the 2.2 runtime packages in our CI runs ... I wish we can figure this out.

This line of script gets the `Major.Minor` of the sdk version, not the runtime version, which is not what we really want (not future-proof). Maybe explicitly use `2.1*` for now (which is also not future-proof, but at least intuitive than the regex matching)?

Reply:
Perhaps we should rename the variable to `$dotnetSdkVersion`.  Keeping this regex would mean that when we move to 2.2, but there's a 2.3 preview out we won't be broken.

Reply:
It also means that when we move to sdk 2.2 but still building against runtime 2.1 (similar to what we have today: sdk 2.1.4, runtime 2.0), we will be broken.

Reply:
Using `dotnet --list-runtimes` shows `Microsoft.NETCore.App 2.1.0-rc1-26423-06` which matches the runtime version of crossgen we need.  So perhaps we can use `dotnet --list-runtimes` and select the newest one and match that runtime version to find appropriate crossgen.

Reply:
We can use `$Script:Options.Framework` to get the runtime version. The framework will always be `netcoreapp2.1` until we update it again.
```
$dotnetRuntimeVersion = $script:Options.Framework -replace 'netcoreapp'
```

Reply:
@daxian-dbw 
> What is not working for preview versions?

I meant that .Net Core 2.1 has new model and always uses (will use) latest runtime versions with `dotnet publish` but it is not working for preview versions.